doc/melange.rst

```diff
@@ -0,0 +1,288 @@
+  ``(runtime_deps ./path/to/file.css (glob_file_rec ./fonts/*))``.
+
+- ``(emit_stdlib <bool>)`` allows the user to specify whether the Melange
+  standard library should be included as a dependency of the stanza or not. The
+  default is ``true``. If this option is ``false``, the Melange standard library
+  and runtime JavaScript files won't be produced in the target directory.
+
+- ``(promote <options>)`` promotes the generated ``.js`` files to the
+  source tree. The options are the same as for the
+  :ref:`rule promote mode <promote>`.
+  Adding ``(promote (until-clean))`` to a ``melange.emit`` stanza will cause
+  Dune to copy the ``.js`` files to the source tree and ``dune clean`` to
+  delete them.
+
+- ``(preprocess <preprocess-spec>)`` specifies how to preprocess files when
+  needed. The default is ``no_preprocessing``. Additional options are described
+  in the :doc:`reference/preprocessing-spec` section.
+
+- ``(preprocessor_deps (<deps-conf list>))`` specifies extra preprocessor
+  dependencies, e.g., if the preprocessor reads a generated file.
+  The dependency specification is described in the :doc:`concepts/dependency-spec`
+  section.
+
+- ``(compile_flags <flags>)`` specifies compilation flags specific to
+  ``melc``, the main Melange executable.
+  ``<flags>`` is described in detail in the
+  :doc:`reference/ordered-set-language` section. It also supports
+  ``(:include ...)`` forms. The value for this field can also be taken
+  from ``env`` stanzas. It's therefore recommended to add flags
+  with e.g. ``(compile_flags :standard <my options>)`` rather than
+  replace them.
+
+- ``(root_module <module>)`` specifies a ``root_module`` that collects all
+  listed dependencies in ``libraries``. See the documentation for
+  ``root_module`` in the :ref:`library` stanza.
+
+- ``(allow_overlapping_dependencies)`` is the same as the corresponding field of
+  :ref:`library`.
+
+Recommended Practices
+=====================
+
+Keep Bundles Small by Reducing the Number of ``melange.emit`` Stanzas
+---------------------------------------------------------------------
+
+It is recommended to minimize the number of ``melange.emit`` stanzas
+that a project defines: using multiple ``melange.emit`` stanzas will cause
+multiple copies of the JavaScript files to be generated if the same libraries
+are used across them. As an example:
+
+.. code:: dune
+
+  (melange.emit
+   (target app1)
+   (libraries foo))
+
+  (melange.emit
+   (target app2)
+   (libraries foo))
+   
+The JavaScript artifacts for library ``foo`` will be emitted twice in the 
+``_build`` folder. They will be present under ``_build/default/app1``
+and ``_build/default/app2``.
+
+This can have unexpected impact on bundle size when using tools like Webpack or
+Esbuild, as these tools will not be able to see shared library code as such,
+as it would be replicated across the paths of the different stanzas
+``target`` folders.
+
+
+Faster Builds with ``subdir`` and ``dirs`` Stanzas
```

Comment:
@christinerose should the "with" be capitalised here? Further up that is the case, I guess we should be consistent but I can never remember which words get capitalised in title case.

Reply:
Strangely, with is capitalized in titles in several style guides, but I double checked, and it's not capitalized in the MLA style guide, which is the one we use if something isn't defined in ours! I'll have to remember that moving forward (as I've capitalized several in reviews!) Perhaps we can pick one for our style guide. Which do you prefer in titles? Capitalized or no?

Reply:
I think I generally prefer capitalised in title case, keeping the number of exceptions minimal. For me this is just `and` and `or`.

Reply:
Capitalised in https://github.com/ocaml/dune/pull/7477/commits/11d11fbb2c34486a0c63ee55099c070b9038a4f0.